python-package/lightgbm/callback.py

```diff
@@ -214,7 +215,24 @@ def _callback(env):
             _init(env)
         if not enabled[0]:
             return
+        if first_metric_only:
+            eval_metric = None
+            for metric_alias in ['metric', 'metrics', 'metric_types']:
+                if metric_alias in env.params.keys():
+                    if isinstance(env.params[metric_alias], (tuple, list)):
+                        eval_metric = env.params[metric_alias][0]
+                    else:
+                        eval_metric = env.params[metric_alias]
+                    break
+            if eval_metric is None:
+                raise LightGBMError("`metric` should be specified if first_metric_only==True.")
```

Comment:
@matsuken92 Can you please explain why this exception is needed? If we don't have any alias of `metric` in params then only default metric for corresponding objective is used. `first_metric_only` makes no sense in this case, just check the first and only metric. For the case of combination of default metric for corresponding objective and custom `feval` function we have guard which performs earlier:
```
if getattr(cb, 'first_metric_only', False) and feval is not None:
    raise LightGBMError("`first_metric_only` and `feval` are not available at the same time.")
```
What am I missing?

Reply:
Oh, you're right, I can remove this `LightGBMError` , if I add one more criteria for `eval_metric is None`.
https://github.com/microsoft/LightGBM/blob/6f30b819db68fc3f1eb03a5d1f27d1ac59769b9a/python-package/lightgbm/callback.py#L234

Reply:
@matsuken92 
Thanks!

Reply:
Error criteria `if eval_metric is None` is now deleted.

Reply:
Also added one more if condition.
https://github.com/microsoft/LightGBM/pull/2209/files#diff-8d4170e1bbb6e2314fc68cc119392b20R232

Reply:
> Error criteria if eval_metric is None is now deleted.

Please add a test for that: `'objective': 'regression', 'first_metric_only': True, 'metric': None` should use default metric for early stopping.